api/app/src/main/kotlin/packit/clients/GithubUserClient.kt

```diff
@@ -0,0 +1,86 @@
+package packit.clients
+
+import org.kohsuke.github.*
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import packit.AppConfig
+import packit.exceptions.PackitAuthenticationException
+import packit.model.User
+import packit.security.Role
+import packit.security.profile.UserPrincipal
+
+@Component
+class GithubUserClient(private val config: AppConfig, private val githubBuilder: GitHubBuilder = GitHubBuilder()) {
+
+    private var github: GitHub? = null
+    private var ghUser: GHMyself? = null
+    private val allowedOrgs = config.authGithubAPIOrgs.split(",")
+
+    fun authenticate(token: String)
+    {
+        connectToClient(token)
+        ghUser = getGitHubUser()
+    }
+
+    fun getUser(): UserPrincipal
+    {
+        checkAuthenticated()
+        val ghu = ghUser!!
+
+        val user = User(
+            1L,
+            ghu.login,
+            "",
+            Role.USER,
+            ghu.name ?: "",
+            mutableMapOf()
+        )
+        return UserPrincipal.create(user, mutableMapOf())
+    }
+
+    fun checkGithubOrgMembership()
+    {
+        checkAuthenticated()
+
+        val userOrgs = ghUser!!.allOrganizations.map {org -> org.login}
+        val inAllowedOrg = allowedOrgs.any {allowed -> userOrgs.contains(allowed)}
```

Comment:
Currently supporting membership of multiple orgs, but as discussed, will replace this in next ticket with single allowed org, with optional team membership